File: app/src/main/java/com/example/mydream/SellerLoginActivity.java
package com.example.mydream;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import java.util.List;

public class SellerLoginActivity extends AppCompatActivity {


    Button login,sign_up;
    EditText user_name,password;
    private SellerViewModel sellerViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_seller_login);
        setId();

        sellerViewModel= new ViewModelProvider(this).get(SellerViewModel.class);
        login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                List<String> check=sellerViewModel.getAllUserName();
                int check1=0;
                for(int i=0;i<check.size();i++){
                    String str1=check.get(i);
                    String str2=user_name.getText().toString();
                    if(str1.equals(str2)){
                        check1=1;
                        break;
                    }
                }
                if(check1==0){
                    Toast.makeText(SellerLoginActivity.this,"This Username Does Not Exist", Toast.LENGTH_SHORT).show();
                    return;
                }
                String str2=password.getText().toString();
                String str1= sellerViewModel.getPassword(user_name.getText().toString());
                if(str1.equals(str2)){
                    Intent it=new Intent(SellerLoginActivity.this, SellerDetailActivity.class);
                    it.putExtra("id",user_name.getText().toString());
                    startActivity(it);
                }
                else{
                    Toast.makeText(SellerLoginActivity.this,"password is incorrect!!!",Toast.LENGTH_SHORT).show();
                }
            }
        });
        sign_up.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent it=new Intent(SellerLoginActivity.this, SellerFormActivity.class);
                startActivity(it);
            }
        });
    }
    public void setId(){
        user_name=findViewById(R.id.login_username);
        password=findViewById(R.id.login_password);
        login=findViewById(R.id.login_button);
        sign_up=findViewById(R.id.signup_button);
    }
}

File: app/src/main/java/com/example/mydream/SellerLoginActivity.kt
package com.example.mydream

import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import com.example.mydream.SellerViewModel
import android.os.Bundle
import com.example.mydream.R
import androidx.lifecycle.ViewModelProvider
import android.content.Intent
import android.widget.Button
import com.example.mydream.SellerDetailActivity
import android.widget.Toast
import com.example.mydream.SellerFormActivity
import kotlinx.coroutines.*
import kotlinx.coroutines.Dispatchers.IO
import kotlinx.coroutines.Dispatchers.Main
import java.sql.DriverManager.println

class SellerLoginActivity : AppCompatActivity() {
    var login: Button? = null
    var sign_up: Button? = null
    var user_name: EditText? = null
    var password: EditText? = null
    private var sellerViewModel: SellerViewModel? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_seller_login)
        setId()
        sellerViewModel = ViewModelProvider(this).get(SellerViewModel::class.java)
        login!!.setOnClickListener {
            CoroutineScope(IO).launch {

                val check = sellerViewModel!!.allUserName
                val check1 = 0
                val str2 = password!!.text.toString()
                val str1 = async {  sellerViewModel!!.getPassword(user_name!!.text.toString())}
                if (str1.await().equals(str2) ) {
                    withContext(Main) {
                        val it = Intent(this@SellerLoginActivity, SellerDetailActivity::class.java)
                        it.putExtra("id", user_name!!.text.toString())
                        Toast.makeText(this@SellerLoginActivity, "password is correct!!!", Toast.LENGTH_SHORT).show()
                        startActivity(it)
                    }
                } else {
                    withContext(Main) {
                        Toast.makeText(this@SellerLoginActivity, "password is incorrecttt!!!", Toast.LENGTH_SHORT).show()
                    }
                }
            }
        }
        sign_up!!.setOnClickListener {
            val it = Intent(this@SellerLoginActivity, SellerFormActivity::class.java)
            startActivity(it)
        }
    }

    fun setId() {
        user_name = findViewById(R.id.login_username)
        password = findViewById(R.id.login_password)
        login = findViewById(R.id.login_button)
        sign_up = findViewById(R.id.signup_button)
    }
}